### PR TITLE
#9978: Add 360 photo support with Mapillary

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
- adding the mapillary viewer for 360 photo support beside Google amd cyclomedia
- handle view custom 360 photos into mapillary viewer using GeoJSONDataProvider
- handle view the custom images in 2d and 3d modes
- install mapillary-js npm package
- edit VectorLayer to enable add layer features via geojson url in 2d + 3d views
- adding unit tests for mapillary

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#9978 

**What is the current behavior?**
#9978 

**What is the new behavior?**
360 Photo support with mapillary viewer is added. It currently is to view only the custom 360 images provided by client not the default mapillary images. User can add the ApiURL parameter into cfg of the streetView plugin in the created context.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
360 Photo support for mapillary is added just for the custom 360 images that can be provided by the client not the default mapillary provider.
- Adding the default mapillary provider is applicable for 2D by adding vector tile support into openLayers [it is not supported in MS] and it needs investigation to handle it in 3D as there is no straight forward support for vector tile layers into Cesium.
- I have added a custom logic to enable vector layer to get url of a geojson file to load its features into the layer for openLayers and Cesium as well.

### CODE REVIEW

File: web/client/components/map/openlayers/Layer.jsx [hunk 1]
```diff
@@ -237,6 +237,15 @@ export default class OpenlayersLayer extends React.Component {
...
             }
+            if (options?.url && options?.isGeojson) {
+                this.layer.getSource().on('featuresloadend', () => {
+                    let extent = this.layer.getSource().getExtent();
+                    if (extent) {
+                        let extendedExtent = [extent[0] - 100, extent[1] - 100, extent[2] + 100, extent[3] + 100];
+                        this.props.map.getView().fit(extendedExtent);
+                    }
+                });
+            }
```
Comment (allyoucanmap) [inline web/client/components/map/openlayers/Layer.jsx:248]: Please remove this changes, if we want to manage a zoom to we should use a zoomTo action in the mapillary viewer component or related files

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/plugins/StreetView/epics/streetView.js [hunk 2]
```diff
@@ -91,6 +93,11 @@ export const streetViewSetupTearDown = (action$, {getState = ()=>{}}) =>
         .filter(() => enabledSelector(getState()))
         .switchMap(() => {
             // setup
+            let streetViewConfigration = streetViewConfigurationSelector(getState());
+            let streetViewDataLayer = { ...streetViewDataLayerSelector(getState()) };
+            if (streetViewConfigration?.provider === 'mapillary') {
+                streetViewDataLayer = { ...streetViewDataLayer, url: streetViewConfigration?.providerSettings?.ApiURL, type: streetViewConfigration?.providerSettings?.type || 'vector' };
+            }
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/plugins/StreetView/epics/streetView.js:100]: what do you mean with 'if the service is not available' ? @allyoucanmap

File: web/client/plugins/StreetView/StreetView.jsx [hunk 3]
```diff
@@ -74,7 +80,7 @@ export default createPlugin(
     'StreetView',
     {
         options: {
-            disablePluginIf: "{state('mapType') === 'leaflet' || state('mapType') === 'cesium'}"
+            disablePluginIf: "{state('mapType') === 'leaflet' || (state('mapType') === 'cesium' && (state('streetView')?.configuration?.provider !== 'mapillary'))}"
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/plugins/StreetView/StreetView.jsx]: Did you solved at the end, with a wrapper?

**Review by tdipisa — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo just merged this. Please test this in DEV asap.

**Pull request merged**